### PR TITLE
Add metacardLocation

### PR DESCRIPTION
**Pull request by emmberk**
This PR is ready for review.

The metacardLocation is needed in the query results, so it must be included in the index request.

### CODE REVIEW

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml [hunk 1]
```diff
@@ -156,6 +157,12 @@ components:
           type: string
           format: url
           example: https://www.example.com/dataset/30f14c6c1fc85cba12bfd093aa8f90e3/file
+        metacardLocation:
+          description: >
+            This URL must support HTTP GET to retrieve the metacard.
+          type: string
+          format: url
+          example: https://www.example.com/dataset/30f14c6c1fc85cba12bfd093aa8f90e3/metacard
```
Comment (peterhuffer) [inline rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: ❓ Should we update these to be UUIDs?

Comment (peterhuffer) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: I guess only the DatasetId path param matters.

Comment (emmberk) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: I'll create another PR right now to update both the datasetId and the examples.

Comment (emmberk) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: https://github.com/connexta/ion-index-api/pull/37

**Comment by emmberk** [comment]
I've heroed this locally.

**Pull request merged**